analyzers/src/SonarAnalyzer.Common/Rules/ShouldImplementExportedInterfacesBase.cs

```diff
@@ -27,86 +27,109 @@
+                            exportedType.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat),
+                            attributeTargetType.ToDisplayString(SymbolDisplayFormat.MinimallyQualifiedFormat)));
+                    }
+                },
+                SyntaxKinds);
+
+        private static bool IsOfExportType(ITypeSymbol type, INamedTypeSymbol exportedType) =>
             type.GetSelfAndBaseTypes()
                 .Union(type.AllInterfaces)
                 .Any(currentType => currentType.Equals(exportedType));
 
-        protected INamedTypeSymbol GetExportedTypeSymbol(SeparatedSyntaxList<TArgumentSyntax>? attributeArguments,
-            SemanticModel semanticModel)
+        private INamedTypeSymbol GetExportedTypeSymbol(SeparatedSyntaxList<TArgumentSyntax>? attributeArguments, SemanticModel semanticModel)
         {
             if (!attributeArguments.HasValue)
             {
                 return null;
             }
 
             var arguments = attributeArguments.Value;
-            if (arguments.Count == 0 ||
-                arguments.Count > 2)
+            if (arguments.Count != 1 && arguments.Count != 2)
             {
                 return null;
             }
 
-            var argumentSyntax = GetArgumentFromNamedArgument(arguments) ??
-                GetArgumentFromSingleArgumentAttribute(arguments) ??
-                GetArgumentFromDoubleArgumentAttribute(arguments, semanticModel);
+            var argumentSyntax = GetArgumentFromNamedArgument(arguments)
+                                 ?? GetArgumentFromSingleArgumentAttribute(arguments)
+                                 ?? GetArgumentFromDoubleArgumentAttribute(arguments, semanticModel);
+
             var typeOfOrGetTypeExpression = GetExpression(argumentSyntax);
 
             var exportedTypeSyntax = GetTypeOfOrGetTypeExpression(typeOfOrGetTypeExpression);
-            if (exportedTypeSyntax == null)
-            {
-                return null;
-            }
-
-            return semanticModel.GetSymbolInfo(exportedTypeSyntax).Symbol as INamedTypeSymbol;
+            return exportedTypeSyntax == null ? null : semanticModel.GetSymbolInfo(exportedTypeSyntax).Symbol as INamedTypeSymbol;
         }
 
+        private ITypeSymbol GetAttributeTargetSymbol(SyntaxNode syntaxNode, SemanticModel semanticModel) =>
+            // Parent is AttributeListSyntax, we handle only class attributes
+            !IsClassOrRecordSyntax(syntaxNode.Parent?.Parent) ? null : semanticModel.GetDeclaredSymbol(syntaxNode.Parent.Parent) as ITypeSymbol;
+
         private TArgumentSyntax GetArgumentFromNamedArgument(IEnumerable<TArgumentSyntax> arguments) =>
             // it's ok to use case insensitive even for C# because if that casing is incorrect the code won't compile
             arguments.FirstOrDefault(x => "contractType".Equals(GetIdentifier(x), StringComparison.OrdinalIgnoreCase));
```

Comment:
If I understand it correctly we can use  Language.NameComparison instead of OrdinalIgnorCase and then we can remove the comment from the line above.